.github/workflows/deploy-storybook.yml

```diff
@@ -0,0 +1,37 @@
+name: Deploy Storybook
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+jobs:
+  deploy:
+    name: Deploy Storybook
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: '14'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Checkout GitHub Pages branch
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "github-actions@github.com"
+          git checkout -B gh-pages
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Build Storybook
+        run: npm run build-storybook
+
+      - name: Push Storybook to GitHub Pages branch
+        run: |
+          git add storybook-static
+          git commit -m "Build Storybook (auto-generated commit)"
+          git push --set-upstream origin gh-pages --force
```

Comment:
Nit: The `--set-upstream` is not useful in an automated job like this, since this entire local checkout will get deleted when the GH Actions job finishes running. It's not harmful, but it might be simpler to just omit it.

Reply:
Done